### PR TITLE
Add SQL MI to quickstarts

### PR DESCRIPTION
**Pull request by srdan-bozovic-msft**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* Added deployment of Azure Sql Database Managed Instance (SQL MI) inside new Virtual Network

## Important
SQL MI is still in gated public preview and it's deployment takes more than 24h. For this reason .ci_skip file is used

### CODE REVIEW

File: 101-sqlmi-new-vnet/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Azure Sql Database Managed Instance (SQL MI) Creation inside New Virtual Network
```
Comment (bmoore-msft) [inline 101-sqlmi-new-vnet/README.md]: Why are we skipping CI?

File: 101-sqlmi-new-vnet/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    "parameters": {
+        "managedInstanceName": {
+            "type": "string",
+            "metadata": {
+                "description": "Enter managed instance name."
+            }
+        },
+        "administratorLogin": {
+            "type": "string",
+            "defaultValue": "cloudSA",
```
Comment (bmoore-msft) [inline 101-sqlmi-new-vnet/azuredeploy.json]: remove defaultValue

File: 101-sqlmi-new-vnet/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+        },
+        "administratorLoginPassword": {
+            "type": "securestring",
+            "metadata": {
+                "description": "Enter password."
+            }
+        },
+        "location": {
+            "type": "string",
+            "defaultValue": "",
```
Comment (bmoore-msft) [inline 101-sqlmi-new-vnet/azuredeploy.json]: use
[resourceGroup().location]
for the defaultValue

File: 101-sqlmi-new-vnet/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+            "metadata": {
+                "description": "Enter location. If you leave this field blank resource group location would be used."
+            }
+        },
+        "virtualNetworkName": {
+            "type": "string",
+            "defaultValue": "",
```
Comment (bmoore-msft) [inline 101-sqlmi-new-vnet/azuredeploy.json]: use a valid/non-empty defaultValue

File: 101-sqlmi-new-vnet/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+        "subnetPrefix": {
+            "type": "string",
+            "defaultValue": "10.0.0.0/24",
+            "metadata": {
+                "description": "Enter subnet address prefix."
+            }
+        },
+        "routeTableName": {
+            "type": "string",
+            "defaultValue": "",
```
Comment (bmoore-msft) [inline 101-sqlmi-new-vnet/azuredeploy.json]: use a valid/non-empty defaultValue

File: 101-sqlmi-new-vnet/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+            "metadata": {
+                "description": "Enter route table name. If you leave this field blank name will be created by the template."
+            }
+        },
+        "skuName": {
```
Comment (bmoore-msft) [inline 101-sqlmi-new-vnet/azuredeploy.json]: move this to a variable, or set some allowedValues

File: 101-sqlmi-new-vnet/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+            "type": "string",
+            "defaultValue": "GP_Gen4",
+            "metadata": {
+                "description": "Enter sku name."
+            }
+        },
+        "storageSizeInGB": {
```
Comment (bmoore-msft) [inline 101-sqlmi-new-vnet/azuredeploy.json]: is there a min/max here?

File: 101-sqlmi-new-vnet/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+            "type": "int",
+            "defaultValue": 32,
+            "metadata": {
+                "description": "Enter storage size."
+            }
+        },
+        "vCores": {
```
Comment (bmoore-msft) [inline 101-sqlmi-new-vnet/azuredeploy.json]: min/max?

File: 101-sqlmi-new-vnet/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+            "type": "int",
+            "defaultValue": 16,
+            "metadata": {
+                "description": "Enter number of vCores."
+            }
+        },
+        "licenseType": {
+            "type": "string",
+            "defaultValue": "BasePrice",
```
Comment (bmoore-msft) [inline 101-sqlmi-new-vnet/azuredeploy.json]: move to variable or use allowedValues

File: 101-sqlmi-new-vnet/azuredeploy.json [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+            "metadata": {
+                "description": "Enter license type."
+            }
+        }
+    },
+    "variables": {
+        "virtualNetworkName": "[if(equals(parameters('virtualNetworkName'),''), concat('vnet-', parameters('managedInstanceName')), parameters('virtualNetworkName'))]",
```
Comment (bmoore-msft) [inline 101-sqlmi-new-vnet/azuredeploy.json]: remove all these conditionals and use the defaultValues defined for the param

File: 101-sqlmi-new-vnet/azuredeploy.json [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+            },
+            "properties": {
+                "administratorLogin": "[parameters('administratorLogin')]",
+                "administratorLoginPassword": "[parameters('administratorLoginPassword')]",
+                "subnetId": "[concat(resourceId('Microsoft.Network/virtualNetworks', variables('virtualNetworkName')),'/subnets/', parameters('subnetName'))]",
+                "storageSizeInGB": "[parameters('storageSizeInGB')]",
+                "vCores": "[parameters('vCores')]",
+                "licenseType": "[parameters('licenseType')]"
+            },
+            "type": "Microsoft.Sql/managedInstances"
```
Comment (bmoore-msft) [inline 101-sqlmi-new-vnet/azuredeploy.json]: move this property to the top

File: 101-sqlmi-new-vnet/azuredeploy.json [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+            },
+            "location": "[variables('location')]",
+            "name": "[parameters('managedInstanceName')]",
+            "sku": {
+                "name": "[parameters('skuName')]"
+            },
+            "properties": {
+                "administratorLogin": "[parameters('administratorLogin')]",
+                "administratorLoginPassword": "[parameters('administratorLoginPassword')]",
+                "subnetId": "[concat(resourceId('Microsoft.Network/virtualNetworks', variables('virtualNetworkName')),'/subnets/', parameters('subnetName'))]",
```
Comment (bmoore-msft) [inline 101-sqlmi-new-vnet/azuredeploy.json]: simplify, just use resourceId()

File: 101-sqlmi-new-vnet/README.md (hunk 1 above)
Comment (srdan-bozovic-msft) [reply 101-sqlmi-new-vnet/README.md]: SQL MI is still in gated public preview so the subscription needs to be whitelisted for deployment. Deployment takes more than 1-2 days.

File: 101-sqlmi-new-vnet/azuredeploy.json (hunk 4 above)
Comment (srdan-bozovic-msft) [reply 101-sqlmi-new-vnet/azuredeploy.json]: Is there a way to leave user a choice to fill in the name or use arm function to generate function instead? ARM functions don't pass validation for this field on the portal?

File: 101-sqlmi-new-vnet/README.md (hunk 1 above)
Comment (bmoore-msft) [reply 101-sqlmi-new-vnet/README.md]: Whitelist the CI subscription and manually validate the deployment

File: 101-sqlmi-new-vnet/azuredeploy.json (hunk 4 above)
Comment (bmoore-msft) [reply 101-sqlmi-new-vnet/azuredeploy.json]: You can just add a defaultValue, e.g.

"defaultValue": "virtualNetwork"

File: 101-sqlmi-new-vnet/README.md (hunk 1 above)
Comment (srdan-bozovic-msft) [reply 101-sqlmi-new-vnet/README.md]: Validated deployment manually.

File: 101-sqlmi-new-vnet/azuredeploy.json (hunk 2 above)
Comment (srdan-bozovic-msft) [reply 101-sqlmi-new-vnet/azuredeploy.json]: removed

File: 101-sqlmi-new-vnet/azuredeploy.json (hunk 3 above)
Comment (srdan-bozovic-msft) [reply 101-sqlmi-new-vnet/azuredeploy.json]: done

File: 101-sqlmi-new-vnet/azuredeploy.json (hunk 4 above)
Comment (srdan-bozovic-msft) [reply 101-sqlmi-new-vnet/azuredeploy.json]: changed to SQLMI-VNET fixed value

File: 101-sqlmi-new-vnet/azuredeploy.json (hunk 7 above)
Comment (srdan-bozovic-msft) [reply 101-sqlmi-new-vnet/azuredeploy.json]: yes, updated

File: 101-sqlmi-new-vnet/azuredeploy.json (hunk 8 above)
Comment (srdan-bozovic-msft) [reply 101-sqlmi-new-vnet/azuredeploy.json]: entered allowed values

File: 101-sqlmi-new-vnet/azuredeploy.json (hunk 9 above)
Comment (srdan-bozovic-msft) [reply 101-sqlmi-new-vnet/azuredeploy.json]: done

File: 101-sqlmi-new-vnet/azuredeploy.json (hunk 10 above)
Comment (srdan-bozovic-msft) [reply 101-sqlmi-new-vnet/azuredeploy.json]: removed vnet and location. route table left as generated but without condition

File: 101-sqlmi-new-vnet/azuredeploy.json (hunk 11 above)
Comment (srdan-bozovic-msft) [reply 101-sqlmi-new-vnet/azuredeploy.json]: moved

**Comment by srdan-bozovic-msft** [comment]
Updated template per request changes

File: 101-sqlmi-new-vnet/README.md [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+</a>
+<a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2Fmaster%2Fazure-quickstart-templates%2Fmaster%2Fazuredeploy.json" target="_blank">
+    <img src="http://armviz.io/visualizebutton.png"/>
+</a>
+
+This template creates a new VNet and deploys a **SQL MI** inside. For **SQL MI** overview visit: https://docs.microsoft.com/en-us/azure/sql-database/sql-database-managed-instance
+
+## Important
+**SQL MI** is still in gated public preview. Before deploying this template you have to whitelist your subscription as explained here: https://docs.microsoft.com/en-us/azure/sql-database/sql-database-managed-instance-create-tutorial-portal#whitelist-your-subscription 
+
```
Comment (bmoore-msft) [inline 101-sqlmi-new-vnet/README.md]: Ok, add a note here that during public preview deployment may take 1-2 days to finish...

File: 101-sqlmi-new-vnet/azuredeploy.json [hunk 14]
```diff
@@ -0,0 +1,177 @@
...
+            },
+            "location": "[parameters('location')]",
+            "name": "[parameters('managedInstanceName')]",
+            "sku": {
+                "name": "[parameters('skuName')]"
+            },
+            "properties": {
+                "administratorLogin": "[parameters('administratorLogin')]",
+                "administratorLoginPassword": "[parameters('administratorLoginPassword')]",
+                "subnetId": "[resourceId(resourceGroup().name, 'Microsoft.Network/virtualNetworks/subnets', parameters('virtualNetworkName'), parameters('subnetName'))]",
```
Comment (bmoore-msft) [inline 101-sqlmi-new-vnet/azuredeploy.json]: simplify: 

resourceId('Microsoft.Network/virtualNetworks/subnets', parameters('virtualNetworkName'), parameters('subnetName'))

File: 101-sqlmi-new-vnet/README.md (hunk 1 above)
Comment (bmoore-msft) [reply 101-sqlmi-new-vnet/README.md]: ok. lmk when the sub is whitelisted and I will test manually

**Pull request merged**